OneCommunityGlobal/HighestGoodNetworkApp
# Shihao - New Feature to Control Display of Badge & Fix Weekly Summaries Report Loading Issue & ESLint Errors

# Description
Fix the issue of failing to load the weekly summaries report page and related eslint errors.
New feature: Add a button 'Load Badges' to control the display of badges

## Related PRS (if any):
This PR is to fix all the issues caused by #1172 and #1242 

## How to test:
**Test ESlint**
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. go to dashboard→ Report → Weekly Summaires Report
5. Make sure the page can load normally and fast
6. Set up ESLint and Prettier in VSCode
7. Check all the files contain no errors in `src/component/WeeklySummariesReport/**`

**Test new features**
1. Click the button 'Load Badges' to show the badge
2. Check if the badge is showing correctly (not everyone has the badge, go find someone who satisfies the condition)

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/1f7791bb-ab9f-439e-86f4-e803edf7bf35
